sdk/src/types/block/core/basic.rs

```diff
@@ -0,0 +1,233 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use packable::{
+    error::{UnpackError, UnpackErrorExt},
+    packer::Packer,
+    unpacker::Unpacker,
+    Packable,
+};
+
+use crate::types::block::{
+    core::{verify_parents, Block},
+    parent::{ShallowLikeParents, StrongParents, WeakParents},
+    payload::{OptionalPayload, Payload},
+    protocol::ProtocolParameters,
+    Error,
+};
+
+/// A builder for a [`BasicBlock`].
+pub struct BasicBlockBuilder {
+    strong_parents: StrongParents,
+    weak_parents: WeakParents,
+    shallow_like_parents: ShallowLikeParents,
+    payload: OptionalPayload,
+    burned_mana: u64,
+}
+
+impl BasicBlockBuilder {
+    /// Creates a new [`BasicBlockBuilder`].
+    #[inline(always)]
+    pub fn new(strong_parents: StrongParents) -> Self {
+        Self {
+            strong_parents,
+            weak_parents: WeakParents::default(),
+            shallow_like_parents: ShallowLikeParents::default(),
+            payload: OptionalPayload::default(),
+            burned_mana: 0,
```

Comment:
Isn't this also required and should be part of the new method?

Reply:
Honestly I feel like we can't really answer these questions until we work on mana management, I don't really know yet how it plays out in the whole workflow. But I'd say you're most probably right so I can change it.